driver-sync/src/main/com/mongodb/client/gridfs/GridFSDownloadStreamImpl.java

```diff
@@ -173,7 +173,7 @@ public synchronized void reset() {
         if (markChunkIndex != chunkIndex) {
             chunkIndex = markChunkIndex;
             buffer = null;
-            cursor = null;
+            discardCursor();
```

Comment:
This was a bug - the cursor wasn't correctly discarded (closed) in the sync driver.